mirror.py

```diff
@@ -29,7 +29,7 @@ class CoverImageMirror(object):
     def __init__(self, db, uploader=None):
         self._db = db
         self.data_source = DataSource.lookup(self._db, self.DATA_SOURCE)
-        self.uploader = uploader or S3Uploader()
+        self.uploader = uploader or S3Uploader.from_config(self._db)
```

Comment:
Actually, given how many times you're calling this, I think it'd be better if I added a `required` argument to `from_config`, which you could use instead of checking the result yourself.